x-pack/plugins/infra/server/routes/snapshot/lib/transform_request_to_metrics_api_request.ts

```diff
@@ -15,12 +15,19 @@ import { transformSnapshotMetricsToMetricsAPIMetrics } from './transform_snapsho
 import { META_KEY } from './constants';
 import { SourceOverrides } from './get_nodes';
 
-export const transformRequestToMetricsAPIRequest = async (
-  client: ESSearchClient,
-  source: InfraSource,
-  snapshotRequest: SnapshotRequest,
-  sourceOverrides?: SourceOverrides
-): Promise<MetricsAPIRequest> => {
+export const transformRequestToMetricsAPIRequest = async ({
+  client,
+  source,
+  snapshotRequest,
+  compositeSize,
+  sourceOverrides,
+}: {
+  client: ESSearchClient;
+  source: InfraSource;
+  snapshotRequest: SnapshotRequest;
+  compositeSize: number;
```

Comment:
Why did you decide to have the composite size as a parameter? Doing it like this the composite size is spread through a lot of places instead of being encapsulated only in this file. 
It seems that there is already a way to override this value with `snapshotRequest.overrideCompositeSize` (which I don't know if it's used somewhere...) so I don't see the need for having it as a parameter.

Reply:
the `overrideCompositeSize` is to override it by passing it as part of the request through the API and since this is reading it from the kibana config (config/kibana.yml).  You can read jason's comments in this issue to learn more.  There is access in the request handlers to this values so anything that needs to call `getNodes` will need to pass it through.

Reply:
oh, I understand! Thanks for the clarification 😃